### PR TITLE
WIP: First rev attempt at removing the deprecated cosmetic filter

### PR DESCRIPTION
**Pull request by bsclifton**
Fixes https://github.com/brave/brave-browser/issues/8914

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
1. Right click a page (any page)
2. There should no longer be a `Brave` context menu item

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by bsclifton** [comment]
All JavaScript unit tests pass - built locally and verified nothing is being thrown via stdout. Could you check this out a little more in-depth, @antonok-edm 😄

File: components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts [hunk 1]
```diff
@@ -1,127 +0,0 @@
...
-        break
-      }
-      const url = tab.url
-      if (url === undefined) {
-        break
-      }
-      shieldsPanelActions.contentScriptsLoaded(tabId, url)
-    }
-  }
-})
```
Comment (antonok-edm) [inline components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts:81]: Oops, I never updated the comment. We'll still need the `'hiddenClassIdSelectors'` and `'contentScriptsLoaded'` cases. `'contextMenuOpened'` can go.

Everything else in this file can still be deleted safely.

File: components/brave_extension/extension/brave_extension/background/events.ts [hunk 2]
```diff
@@ -7,7 +7,6 @@ require('./events/tabsEvents')
 require('./events/shieldsEvents')
 require('./events/runtimeEvents')
 require('./events/webNavigationEvents')
-require('./events/cosmeticFilterEvents')
```
Comment (antonok-edm) [inline components/brave_extension/extension/brave_extension/background/events.ts:10]: This is still required, see my comment on it below.

**Comment by antonok-edm** [comment]
By the way, most of the testing for the newer system is actually in C++ browser tests. I'd recommend `npm run test -- brave_browser_tests --filter='Cosmetic*.*'` to verify this hasn't broken anything.

File: components/brave_extension/extension/brave_extension/background/events.ts (hunk 2 above)
Comment (bsclifton) [reply components/brave_extension/extension/brave_extension/background/events.ts]: added back 😄

File: components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts (hunk 1 above)
Comment (bsclifton) [reply components/brave_extension/extension/brave_extension/background/events/cosmeticFilterEvents.ts]: Added back! 😄 👍

**Comment by bsclifton** [comment]
@antonok-edm thanks for the above feedback 😄 Just addressed those items

As for preserving the existing rules, I'm not sure if this was a well-known enough feature to warrant that kind of migration. I'll check with a few folks and circle back

**Comment by pes10k** [comment]
just for my two cents, I second @bsclifton 's idea.  Unless its very simple to migrate them, i think its probably not worth the trouble :)

**Comment by bsclifton** [comment]
OK going to go ahead and close this up, after discussion 😄

**Pull request closed without merging**